Formula/crystal.rb

```diff
@@ -45,6 +45,7 @@ class Crystal < Formula
   depends_on "llvm@14"
   depends_on "openssl@1.1" # std uses it but it's not linked
   depends_on "pcre"
```

Comment:
This dependency should be removed then?

Reply:
It might be best to leave it in for a transition period in order to allow easily switching back to PCRE if needed. This is possible via compile time flag `-Duse_pcre`.

Reply:
Also note that PCRE2 is currently (as of Crystal 1.7) only available as a preview (via `-Duse_pcre2`). It is expected to become the default in 1.8 (and this is already the case for nightly builds). At least until the release of 1.8 PCRE is definitely necessary.

Reply:
So maybe we should wait until 1.8 is released and then make the PCRE switch?

Reply:
The general idea with the opt-in and opt-out mechanisms is to give users more time to apply necessary modifications to their code, in case that's necessary. The transition to PCRE2 is expected to be relatively smooth for most applications, but some use cases may require code changes to support PCRE2.

Adding PCRE2 in advance allows users to test compatibility prior to the release of 1.8.
Keeping PCRE around after 1.8 is released allows users to easily switch back to PCRE to enable builds until errors are fixed.

Reply:
There is no linkage with PCRE2, so why isn't having users install PCRE2 themselves separately enough of an opt-in mechanism?

Reply:
Ah right, good point. I think that should be sufficient for users.

However, *building* the compiler already requires PCRE2 on master. So I'm pretty sure a HEAD build currently fails without the `pcre2` dependency (haven't checked it, but it's supposed to).

Reply:
@straight-shoota Done.